client/blocks/upe/index.js

```diff
@@ -2,11 +2,70 @@ import {
 	registerPaymentMethod,
 	registerExpressPaymentMethod,
 } from '@woocommerce/blocks-registry';
-import upePaymentMethod from './payment-method';
+import { getDeferredIntentCreationUPEFields } from './upe-deferred-intent-creation/payment-elements.js';
+import { SavedTokenHandler } from './saved-token-handler';
 import paymentRequestPaymentMethod from 'wcstripe/blocks/payment-request';
+import WCStripeAPI from 'wcstripe/api';
+import { getBlocksConfiguration } from 'wcstripe/blocks/utils';
+import { getStripeServerData } from 'wcstripe/stripe-utils';
 
 // Register Stripe UPE.
-registerPaymentMethod( upePaymentMethod );
+const upeMethods = {
+	card: 'stripe',
+	bancontact: 'stripe_bancontact',
+	au_becs_debit: 'stripe_au_becs_debit',
+	eps: 'stripe_eps',
+	giropay: 'stripe_giropay',
+	ideal: 'stripe_ideal',
+	p24: 'stripe_p24',
+	sepa_debit: 'stripe_sepa_debit',
+	sofort: 'stripe_sofort',
+	affirm: 'stripe_affirm',
+	afterpay_clearpay: 'stripe_afterpay_clearpay',
+};
+
+const api = new WCStripeAPI(
+	getStripeServerData(),
```

Comment:
Another thing that could point to a compilation problem on my end is that I get a 404 when trying to retrieve `/build/upe_blocks.css` when visiting both the block and the classic checkout pages. Does this happen on your end by any chance?

<img width="2382" alt="image" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/41606954/2f3c47fd-8c6d-46f9-b6ef-3221c292f443">

Reply:
hmm I'm not getting the first `wc_stripe_upe_params` error but I am seeing the `upe_blocks.css` error. 

I did some digging and the issue causing the `upe_blocks.css` error is I no longer engage the `fields.js` file which imported the `styles.css`. 

So, the webpack build no longer traverses that file and so it doesn't build a upe_blocks.css file. 

I can fix it by making sure that I import the `styles.css` but that css file houses Stripe Link styles. 

I haven't looked into Stripe Link yet and whether I just need to import that `styles.css` style sheet in the `index.js`. 🤔 🤷‍♂️

Reply:
> Uncaught ReferenceError: wc_stripe_upe_params is not defined

Do you happen to have a set up where you have block checkout page but the classic checkout is your actual default?

Reply:
> Do you happen to have a set up where you have block checkout page but the classic checkout is your actual default?

I think that's it. I could replicate if I created a second block checkout page.

<img width="637" alt="Screenshot 2023-11-29 at 11 09 51 am" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/8490476/9bfc1416-c635-46b9-bfc0-662481118143">

The issue is `has_cart_or_checkout_on_current_page()` only returns true if the current page is the offical cart or checkout page. Which causes the `payment_scripts()` function to return early and not load the `wc_stripe_upe_params` scripts. 

I should be able to fix it by changing `has_cart_or_checkout_on_current_page()` to also look for checkout/cart blocks on the page.

Reply:
> Uncaught ReferenceError: wc_stripe_upe_params is not defined

Fixed in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2784/commits/f4a7158e469cb039e45cdc479d80303ff4ea7f93

I still don't know what to do about the `/build/upe_blocks.css` error.

Reply:
> I still don't know what to do about the /build/upe_blocks.css error.

I ended up importing the `styles.css` which fixed the `/build/upe_blocks.css error`. 

I verified that without, once I added Link support, the button was un-styled. 

| Before | After |
|--------|--------|
| <img width="523" alt="Screenshot 2023-11-29 at 5 23 42 pm" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/8490476/fb605299-851c-4fee-8d35-3940711d233d"> | <img width="509" alt="Screenshot 2023-11-29 at 5 22 56 pm" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/8490476/9f5e906c-1212-408d-bfad-cf6bca6822b6"> | 

This PR now supports rending Stripe Link and authing with Stripe Link but doesn't currently work in taking payments with Stripe Link. Theres more research I have to do into what I need to do there. It could even be seperate to this PR. 🤷

Reply:
> This PR now supports rending Stripe Link and authing with Stripe Link but doesn't currently work in taking payments with Stripe Link. Theres more research I have to do into what I need to do there. It could even be seperate to this PR.

Nvm I got it working and fixed the mandate issue in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2784/commits/a2c26235b008f2866007a15f07c7fd3001c044bf.